src/packages/core/src/server.ts

```diff
@@ -183,27 +183,24 @@ export class Server extends Emittery<{ open: undefined; close: undefined }> {
           else throw err;
         }
       })
-    ]).catch(async error => {
+    ]).then((promiseResults) => {
+      if (promiseResults[0].status === "fulfilled" && promiseResults[1].status === "fulfilled") {
+        this.emit("open");
+      } else {
+        let reason = "";
+        reason += promiseResults[0].status === "rejected" ? `${promiseResults[0].reason}\n\n` : "";
+        reason += promiseResults[1].status === "rejected" ? promiseResults[1].reason : "";
+        throw reason;
+      }
+    }).catch(async error => {
       this.#status = Status.unknown;
       if (callbackIsFunction) callback!(error);
       await this.close();
       throw error;
     });
```

Comment:
Should this be something like?

```suggestion
    }).catch(error => {
      this.#status = Status.unknown;
	  await this.close();
	  throw error;
    }).catch(error => {
      if (callbackIsFunction) callback!(error);
      else throw error;
    });
```

Alternatively, since `this.close()` could also cause an error, we may want to implement `AggregateError` here, as well as above, but I don't know how AggregateError + AggregateError would behave. OOORRRRRR we ignore `this.close()` errors (`await this.close().catch(()=>{});` since this isn't the error the user would really care about anyway.

Really, though, it feels like this section needs to be refactored a bit to handle these error cases more eloquently. Any thoughts?